core/src/main/java/net/sourceforge/jnlp/util/LocaleUtils.java

```diff
@@ -0,0 +1,88 @@
+package net.sourceforge.jnlp.util;
+
+import java.util.Locale;
+import java.util.Objects;
+import net.adoptopenjdk.icedteaweb.xmlparser.ParseException;
+
+import static net.adoptopenjdk.icedteaweb.i18n.Translator.R;
+
+public class LocaleUtils {
+
+    public enum Match { LANG_COUNTRY_VARIANT, LANG_COUNTRY, LANG, GENERALIZED }
+
+    /**
+     * Returns a {@link Locale} from a locale string. Each locale is specified by a language identifier,
+     * possibly country identifier, and possibly a variant. The syntax is as follows:
+     * <pre>
+     * locale ::= language [ "_" country [ "_" variant ] ]
+     * </pre>
+     *
+     * @param localeString the locale string
+     * @return locale of document
+     */
+    public static Locale getLocale(final String localeString) throws ParseException {
+        if (Objects.isNull(localeString) || localeString.length() < 2) {
+            throw new ParseException(R("PIllegalLocale", localeString));
+        }
+
+        final String language = localeString.substring(0, 2);
+        final String country = (localeString.length() < 5) ? "" : localeString.substring(3, 5);
+        final String variant = (localeString.length() > 7) ? localeString.substring(6) : "";
+
+        // null is not allowed n locale but "" is
```

Comment:
this comment is misleading as there cannot be a null value at this point.
Please remove the comment

Reply:
done